[SYCL][Graph]Use doxygen style comments in graphs header files

EwanC opened the pull request:
As well as the doxygen comments there are some additional changes I found along the way:
* Pass queue by reference to `begin_recording()` & `end_recording()`
* Pass node by reference to `make_edge()`
* Add stubbed out `update()` entry-point
* Remove unused `MParent` variable

EwanC commented on sycl/include/sycl/ext/oneapi/experimental/graph.hpp:
```diff
@@ -108,54 +124,67 @@ class __SYCL_EXPORT command_graph {
   /// @return True if the queue had its state changed from recording to
   /// executing.
-  bool end_recording(queue recordingQueue);
+  bool end_recording(queue &recordingQueue);
```
```suggestion
  bool end_recording(queue &RecordingQueue);
```

Bensuo commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -194,66 +263,93 @@ struct graph_impl {
                   const std::shared_ptr<graph_impl> &GraphImpl)
       : MSchedule(), MGraphImpl(GraphImpl), MPiCommandBuffers(),
         MPiSyncPoints(), MContext(Context) {}
+
+  /// Destructor
```
Nitpick but I think this is probably unnecessary. Having it doesn't necessarily hurt anything other than create a bit of noise I guess, but perhaps better not to add that where possible.

EwanC replied:
Good point, right now the comment doesn't communicate any extra info. Rather than deleting it, I could add to the detailed description that the destructor releases the underlying PI command-buffers, which would give the comment informative value.

Bensuo reviewed (approved):
LGTM

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -147,39 +198,57 @@ struct graph_impl {
 
+  /// Find the sycl event associated with a node
+  /// @param NodeImpl Node to find event for
+  /// @return event associated with node
```
```suggestion
  /// @return Event associated with node
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -194,66 +263,93 @@ struct graph_impl {
   add_subgraph_nodes(const std::list<std::shared_ptr<node_impl>> &NodeList);
+
+  /// Query for the context tied to this graph
+  /// @return context associated with graph
```
```suggestion
  /// @return Context associated with graph
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -194,66 +263,93 @@ struct graph_impl {
   void create_pi_command_buffers(sycl::device D);
 
+  /// Query for the context tied to this graph
+  /// @return context associated with graph
```
```suggestion
  /// @return Context associated with graph
```

julianmi commented on sycl/include/sycl/ext/oneapi/experimental/graph.hpp:
```diff
@@ -84,13 +90,20 @@ class __SYCL_EXPORT node {
+
+  /// Add an empty node to the graph
+  /// @param PropList property list used to pass [0..n] predecessor nodes
+  /// @return constructed empty node which has been added to the graph
```
```suggestion
  /// @return Constructed empty node which has been added to the graph
```

julianmi commented on sycl/include/sycl/ext/oneapi/experimental/graph.hpp:
```diff
@@ -99,7 +112,10 @@ class __SYCL_EXPORT command_graph {
+  /// Add a command-group node to the graph
+  /// @param CGF command group function to create node with
+  /// @param PropList property list used to pass [0..n] predecessor nodes
+  /// @return constructed node which has been added to the graph
```
```suggestion
  /// @return Constructed node which has been added to the graph
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -123,18 +151,41 @@ struct node_impl {
+  void remove_root(const std::shared_ptr<node_impl> &Root);
+
+  /// Create a kernel node in the graph
+  /// @param Kernel kernel to run when node executes
```
```suggestion
  /// @param Kernel Kernel to run when node executes
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -123,18 +151,41 @@ struct node_impl {
+
+  /// Create a kernel node in the graph
+  /// @param Kernel kernel to run when node executes
+  /// @param NDRDesc NDRange Decription for kernel
```
```suggestion
  /// @param NDRDesc NDRange decription for kernel
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -147,39 +198,57 @@ struct graph_impl {
   bool clear_queues();
 
+  /// Associate a sycl event with a node in the graph
+  /// @param EventImpl event to associate with a node in map
```
```suggestion
  /// @param EventImpl Event to associate with a node in map
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -147,39 +198,57 @@ struct graph_impl {
 
+  /// Associate a sycl event with a node in the graph
+  /// @param EventImpl event to associate with a node in map
+  /// @param NodeImpl node to associate with event in map
```
```suggestion
  /// @param NodeImpl Node to associate with event in map
```

julianmi commented on sycl/source/detail/graph_impl.hpp:
```diff
@@ -194,66 +263,93 @@ struct graph_impl {
 class exec_graph_impl {
 public:
+  /// Constructor
+  /// @param Context context to create graph with
```
```suggestion
  /// @param Context Context to create graph with
```

julianmi reviewed (commented):
LGTM. I added some minor style suggestions. We might also think about unifying the use of periods after comments.

EwanC commented:
> LGTM. I added some minor style suggestions. We might also think about unifying the use of periods after comments.

Good idea 👍  I've updated this to consistently use periods.
